server/fleet/activities.go

```diff
@@ -4,48 +4,416 @@ import (
-	// ActivityTypeCreatedPolicy is the activity type for created policies
-	ActivityTypeCreatedPolicy = "created_policy"
-	// ActivityTypeEditedPolicy is the activity type for edited policies
-	ActivityTypeEditedPolicy = "edited_policy"
-	// ActivityTypeDeletedPolicy is the activity type for deleted policies
-	ActivityTypeDeletedPolicy = "deleted_policy"
-	// ActivityTypeAppliedSpecPolicy is the activity type for saved queries spec applied
-	ActivityTypeAppliedSpecPolicy = "applied_spec_policy"
-	// ActivityTypeCreatedSavedQuery is the activity type for created saved queries
-	ActivityTypeCreatedSavedQuery = "created_saved_query"
-	// ActivityTypeEditedSavedQuery is the activity type for edited saved queries
-	ActivityTypeEditedSavedQuery = "edited_saved_query"
-	// ActivityTypeDeletedSavedQuery is the activity type for deleted saved queries
-	ActivityTypeDeletedSavedQuery = "deleted_saved_query"
-	// ActivityTypeDeletedMultipleSavedQuery is the activity type for multiple deleted saved queries
-	ActivityTypeDeletedMultipleSavedQuery = "deleted_multiple_saved_query"
-	// ActivityTypeAppliedSpecSavedQuery is the activity type for saved queries spec applied
-	ActivityTypeAppliedSpecSavedQuery = "applied_spec_saved_query"
-	// ActivityTypeCreatedTeam is the activity type for created team
-	ActivityTypeCreatedTeam = "created_team"
-	// ActivityTypeDeletedTeam is the activity type for deleted team
-	ActivityTypeDeletedTeam = "deleted_team"
-	// ActivityTypeLiveQuery is the activity type for live queries
-	ActivityTypeLiveQuery = "live_query"
-	// ActivityTypeUserAddedBySSO is the activity type for new users added
-	// via SSO JIT provisioning
-	ActivityTypeUserAddedBySSO = "user_added_by_sso"
-	// ActivityTypeEditedAgentOptions is the activity type for when the agent
-	// options are edited (either globally or for a team).
-	ActivityTypeEditedAgentOptions = "edited_agent_options"
-	// ActivityTypeAppliedSpecTeam is the activity type for a team spec applied
-	ActivityTypeAppliedSpecTeam = "applied_spec_team"
-)
+//go:generate go run gen_activity_doc.go ../../docs/Using-Fleet/Audit-Activities.md
+
+// ActivityDetailsList is used to generate documentation.
+var ActivityDetailsList = []ActivityDetails{
+	ActivityTypeCreatedPack{},
+	ActivityTypeEditedPack{},
+	ActivityTypeDeletedPack{},
+	ActivityTypeAppliedSpecPack{},
+
+	ActivityTypeCreatedPolicy{},
+	ActivityTypeEditedPolicy{},
+	ActivityTypeDeletedPolicy{},
+	ActivityTypeAppliedSpecPolicy{},
+	ActivityTypeCreatedSavedQuery{},
+	ActivityTypeEditedSavedQuery{},
+	ActivityTypeDeletedSavedQuery{},
+	ActivityTypeDeletedMultipleSavedQuery{},
+	ActivityTypeAppliedSpecSavedQuery{},
+
+	ActivityTypeCreatedTeam{},
+	ActivityTypeDeletedTeam{},
+	ActivityTypeAppliedSpecTeam{},
+
+	ActivityTypeEditedAgentOptions{},
+
+	ActivityTypeLiveQuery{},
+
+	ActivityTypeUserAddedBySSO{},
+}
+
+type ActivityDetails interface {
+	ActivityName() string
+	Documentation() (activity string, details string, detailsExample string)
+}
```

Comment:
spitballing here not requesting any changes, but what if we namespace using packages and remove `Activity` from the names? a new file in `server/fleet/activity/activities.go` that can be used via:

```go
var a activity.CreatedPack
a.Name()
```